uber/models.py

```diff
@@ -2038,7 +2052,9 @@ class PromoCode(MagModel):
             func.replace(func.replace(func.lower(code), '-', ''), ' ', ''),
             unique=True),
         CheckConstraint(func.trim(code) != '',
-            name='ck_promo_code_non_empty_code'))
+            name='ck_promo_code_non_empty_code'),
+        CheckConstraint(or_(discount != None, uses_allowed != None),
+            name='ck_promo_code_no_unlimited_use_free_badge'))
```

Comment:
This is potentially an awesome change, or a controversial change. This constraint disallows unlimited-use free badge promo codes.

It's still possible to create an unlimited-use promo code that gives free badges by creating a fixed discount greater than the badge price. I couldn't figure out how to prevent that at the database level.